daml/Marketplace/Distribution/Bidding/Request.daml

```diff
@@ -12,14 +17,13 @@ template Request
   with
     operator : Party
     provider : Party
-    issuer : Party
     customer : Party
     auctionId : Text
-    asset : Asset
   where
-    signatory operator, provider, issuer
+    signatory operator, provider, customer
 
-    let assetId = asset.id
+    key (customer, auctionId) : (Party, Text)
+    maintainer key._1
 
     controller customer can
       nonconsuming SubmitBid : ContractId Bidding.Bid
```

Comment:
Ensuring a party only submits one `Bid` is enforced by unique key constraint - A cleaner solution around this could be to add `Optional Bid` which gets set once a bid is submitted. This contract is short lived per `Auction`